packages/directory/addon/controllers/directory.js

```diff
@@ -51,10 +57,14 @@ export default Controller.extend({
   /**
    * @property {String} title - Title for the table
    */
-  title: computed('filter', function() {
-    let title = dirInfo[0].name,
+  title: computed('filter', 'target.currentPath', function() {
+    const currentPath = get(this, 'target.currentPath'),
```

Comment:
Can we use this: 
https://www.emberjs.com/api/ember/3.7/classes/RouterService/properties/currentRoute?anchor=currentRouteName

I think it is a little clearer.

Reply:
why inject a service when the property exists in controllers already?  That's kind of like knocking on your neighbors door so he can tell you what color your own kitchen walls are.

Reply:
If clarity is the issue, I can add a comment to that line to make it more clear.

Reply:
To me, `target` is one of those obscure legacy concepts that would be nice to be cleaned up. 

Just look at the doc definition:

> The object to which actions from the view should be sent.

Nothing indicates "I have knowledge of the current route".

When reading the code the router service feels like a more obvious way to get the current route. It also has better documentation:

https://www.emberjs.com/api/ember/3.7/classes/RouterService/properties/currentRoute?anchor=currentRouteName

Also, the router service is a more portable technique. If we move this logic to component unlike `target` the router service can still be used.

>That's kind of like knocking on your neighbors door so he can tell you what color your own kitchen walls are.

Keep in mind that the target is set (injected) into the controller as well.

Reply:
Is there indication that this is legacy as in going to be deprecated?

The doc for `target` also says: 
> by default, the value of the target property is set to the router

Which does give indication that target knows about the route.

No argument on better documentation.

> Also, the router service is a more portable technique. If we move this logic to component unlike `target` the router service can still be used.

agreed it's more portable, but it's odd that it's injecting a service to get info from the router, when the router is already available in the object.  It's completely redundant.  Router service is nice to have for components and classes that don't already have access to the router. But why inject it when the object already has the router?